specification/quota/resource-manager/Microsoft.Quota/preview/2023-06-01-preview/examples/GroupQuotaLimits/PutGroupQuotaLimits-Compute.json

```diff
@@ -1,51 +1,52 @@
 {
   "operationId": "GroupQuotaLimits_CreateOrUpdate",
   "parameters": {
-    "mgId": "E7EC67B3-7657-4966-BFFC-41EFD36BAA09",
-    "resourceProviderScope": "providers/Microsoft.Compute/locations/westus",
+    "managementGroupId": "E7EC67B3-7657-4966-BFFC-41EFD36BAA09",
     "groupQuotaName": "groupquota1",
     "resourceName": "cores",
     "api-version": "2023-06-01-preview",
```

Comment:
now I look at this again, this seems like a misuse of PUT semantics:
using PUT implies you're *redefining* the cores limit, to have certain scope of Compute+westus.

Reply:
That's correct. We had (/providers/Microsoft.Compute/locations/westUs) as scope. 
But it created 3 providers in URi - 1. Microsoft.ManagementGroups, 2. Micisoft.Compute, 3. Microsoft.Quota.
This created an ARM routing issue to Microsoft.Quota RP, as Mocrosoft.Compute and other RPs in the middle needs to declare Location as extension resource, which doesn't make sense, so that ARM can route the call to the 3rd RP Quota (routingType=CascadeExtension).
That's why ARM Team suggested this solution.
Yes, this API doesn't look good, agreed So, we are considering following:
For Quota and usages APIs, as both goes together (Usages API not included yet in swagger)-
**PUT , Patch - Quota API**
[https://management.azure.com/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Quota/groupQuotas/{groupQuotaName}/resourceProviders/{rpName}/locationLimits/{location}?api-version=2023-06-01-preview] -
There is another ARM restriction to only 3 levels of child resources, so moving the resourceName to request body. (Reference - 
Reference: [Resource Provider Contract(RPC) Guidelines | ARM Wiki](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc030-avoid-excessive-resource-type-nesting) : 
**_ARM only officially supports 3 levels of nested resource types (i.e. {type1}/{name1}/{type2}/{name2}/{type3}/{name3}).
Even if you omit last {name4}, it’s counted as 4 types_**.}

**Request body:**
{
     // Request Body has quota limits across VM Families
“name:{
 “value”: “standardDDV4”
},
“Limit”: 200
}
Similarly Get API for quota-
[https://management.azure.com/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Quota/groupQuotas/{groupQuotaName}/resourceProviders/{rpName}/locationLimits/{location}?api-version=2023-06-01-preview] 

**Usages API:**
GET /providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Quota/groupQuotas/{groupQuotaName}/resourceProviders/{rpName}/locationUsages/{location}?api-version=2023-06-01-preview

Please suggest, what do you think, we are planning to go back to ARM with this proposal, as it seems better than -
. "resourceProviderScope": "providers/Microsoft. Compute/locations/westus", - For Put
And $filter for Get - $filter="provider eq Microsoft.Compute and locations eq westUs"

And it's not a breaking change, as we haven't onboarded any users. So, any change in API is apricated. I forwarded the email thread with ARM team for reference to you. It's good that you are raising this question and let's have a fresh perspective.